skbeam/core/roi.py

```diff
@@ -430,18 +476,27 @@ def circular_average(image, calibrated_center, threshold=0, nx=100,
     ring_averages : array
         Radial average of the image. shape is (nx, ).
     """
-    radial_val = utils.radial_grid(calibrated_center, image.shape, pixel_size)
-
-    bin_edges, sums, counts = utils.bin_1D(np.ravel(radial_val),
-                                           np.ravel(image), nx,
+    radial_val = utils.radial_grid(calibrated_center, image.shape, pixel_size) 
+    if mask is not None:  
+        mask = np.array( mask, dtype = bool)
+        binr = radial_val[mask]
+        image_mask =     np.array( image )[mask]        
+    else:        
+        binr = np.ravel( radial_val ) 
+        image_mask = np.ravel(image) 
+    binr_ =   binr /(np.sqrt(pixel_size[1]*pixel_size[0] ))
+    bin_edges, sums, counts = bin_1D(      binr_,
+                                           image_mask,
+                                           nx=nx,
                                            min_x=min_x,
-                                           max_x=max_x)
+                                           max_x=max_x) 
     th_mask = counts > threshold
```

Comment:
For best usability I think it would be better to remove this line, it assumes that a threshold mask is desired, which may not be the case. If a threshold mask is desired then that could be handled in the mask argument.